src/versionbits.cpp

```diff
@@ -175,6 +175,42 @@ int AbstractThresholdConditionChecker::GetStateSinceHeightFor(const CBlockIndex*
     return pindexPrev->nHeight + 1;
 }
 
+std::vector<SignalInfo> AbstractThresholdConditionChecker::GetSignalInfo(const CBlockIndex* pindex) const
+{
+    std::vector<SignalInfo> result;
+
+    int bip = 0, bip_version = 0;
+    const bool check_other_versions = BIP(bip, bip_version);
+
+    const int32_t activate = ActivateVersion();
+    const int32_t abandon = AbandonVersion();
+    const int period = Period();
+
+    while (pindex != nullptr) {
+        if (pindex->nVersion == activate) {
+            result.push_back({ .height = pindex->nHeight, .bip_version = -1, .activate = true });
+        } else if (pindex->nVersion == abandon) {
+            result.push_back({ .height = pindex->nHeight, .bip_version = -1, .activate = false });
+        } else if (check_other_versions) {
+            if ((pindex->nVersion & 0x00FFFF00l) == (bip << 8)) {
+                SignalInfo s;
+                s.height = pindex->nHeight;
+                s.bip_version = static_cast<uint8_t>(pindex->nVersion & 0xFF);
+                if ((pindex->nVersion & 0xFF000000l) == VERSIONBITS_TOP_ACTIVE) {
+                    s.activate = true;
+                    result.push_back(s);
+                } else if ((pindex->nVersion & 0xFF000000l) == VERSIONBITS_TOP_ABANDON) {
+                    s.activate = false;
+                    result.push_back(s);
+                }
```

Comment:
Is it worth doing something here instead of just dropping `s` on the floor?

Reply:
Wow, not a fan of the "l" (ell) in that font looking quite so much like a "1" (one).

Maybe it could make sense to test it as:

```c++
const uint8_t maybe_version = static_cast<uint8_t>(pindex->nVersion  & 0xFF);
if ((activate >> 8) == (pindex->nVersion >> 8)) {
     result.push_back({.height = pindex->nHeight, .bip_version = maybe_verison, .activate = true});
} else if ((abandon >> 8) == (pindex->nVersion >> 8)) {
     result.push_back({.height = pindex->nHeight, .bip_version = maybe_verison, .activate = false});
}
```

If the top bits aren't activate or abandon, they might just be using bip-9 versionbits signalling or similar, so I think it's better not to try to guess at what they might mean...

Reply:
Now that i'm thinking from the practical angle, i think it's fine either way :)